### DISCUSSION TITLE
Pytorch Lightning: Support of DDPStrategy in PyG

### DISCUSSION THREAD

**Discussion by ZikangZhou** [pyg-team/pytorch_geometric#4665] [Q&A]
Hi PyG team,

I'm trying to use PyTorch Lightning for multi-gpu training. The data in my task are multiple small graphs, not a single large graph. 
I looked at the examples of PyG + PyTorch Lightning and found that all examples use DDPSpawnStrategy instead of DDPStrategy. I'm wondering whether I can switch to DDPStrategy without modifying my code in my case?

**Comment by rusty1s** [comment]
You should be able to use `DDPStrategy` but keep in mind that each GPU process will create a replica of the dataset in CPU memory. Therefore, we generally recommend to use `DDPSpawnStrategy` to be able to leverage shared memory across processes. Another alternative is to not make use of `InMemoryDataset` and only load in the graphs from disk that are actually used within a mini-batch (I/O might be a bottleneck here).
